src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -2161,20 +2165,193 @@ private struct SmallStructWithValueAndReferenceTypes
         {
             public string MyString { get; }
             public int MyInt { get; set; }
+            public bool? MyBool { get; set; }
             public Point_2D_Struct MyPointStruct { get; set; }
 
             [JsonConstructor]
             public SmallStructWithValueAndReferenceTypes(
                 Point_2D_Struct myPointStruct,
                 string myString = "Default",
-                int myInt = -1)
+                int myInt = -1,
+                bool? myBool = true)
             {
                 MyString = myString;
                 MyInt = myInt;
+                MyBool = myBool;
                 MyPointStruct = myPointStruct;
             }
         }
 
-        private class PointClass { }
+        public class PointClass { }
+
+        [Fact]
+        public static void Ignore_WhenWriting_Default_Globally()
+        {
+            var options = new JsonSerializerOptions
+            {
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
+            };
+
+            string json = @"{
+""MyPointClass2"":{},
+""MyString1"":""Default"",
+""MyBool1"":null,
+""MyInt2"":0,
+""MyPointStruct2"":{""X"":1,""Y"":2},
+""MyInt1"":1,
+""MyString2"":null,
+""MyPointClass1"":null,
+""MyBool2"":true,
+""MyPointStruct1"":{""X"":0,""Y"":0}
+}";
+
+            // All members should correspond to JSON contents, as ignore doesn't apply to deserialization.
+            ClassWithThingsToIgnore obj = JsonSerializer.Deserialize<ClassWithThingsToIgnore>(json, options);
+            Assert.NotNull(obj.MyPointClass2);
+            Assert.Equal("Default", obj.MyString1);
+            Assert.Null(obj.MyBool1);
+            Assert.Equal(0, obj.MyInt2);
+            Assert.Equal(1, obj.MyPointStruct2.X);
+            Assert.Equal(2, obj.MyPointStruct2.Y);
+            Assert.Equal(1, obj.MyInt1);
+            Assert.Null(obj.MyString2);
+            Assert.Null(obj.MyPointClass1);
+            Assert.True(obj.MyBool2);
+            Assert.Equal(0, obj.MyPointStruct1.X);
+            Assert.Equal(0, obj.MyPointStruct1.Y);
+
+            // Ignore null as appropriate during serialization.
+            string expectedJson = @"{
+""MyPointClass2"":{},
+""MyString1"":""Default"",
+""MyInt2"":0,
+""MyPointStruct2"":{""X"":1,""Y"":2},
+""MyInt1"":1,
+""MyBool2"":true,
+""MyPointStruct1"":{""X"":0,""Y"":0}
+}";
+            JsonTestHelper.AssertJsonEqual(expectedJson, JsonSerializer.Serialize(obj, options));
+        }
+
+        public class ClassWithThingsToIgnore
+        {
+            public string MyString1 { get; set; }
+            [JsonInclude]
+            public string MyString2;
+            [JsonInclude]
+            public int MyInt1;
```

Comment:
nit: `MyInt2` should be the one with `JsonInclude`.

Reply:
There's no order here - I randomized it so we have null handling coverage for both properties/fields being nullable/non-nullable